yt_dlp/extractor/echo360.py

```diff
@@ -0,0 +1,108 @@
+import re
+
+from urllib.parse import urlparse, urlunparse
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    float_or_none,
+    traverse_obj,
+)
+
+
+class Echo360IE(InfoExtractor):
+    _VALID_URL = r'''(?x)
+        https?://(?P<host>echo360\.(?:ca|net\.au|org|org\.au|org\.uk))/
+        media/(?P<id>[\da-fA-F]{8}-(?:[\da-fA-F]{4}-){3}[\da-fA-F]{12})/public
+    '''
+
+    _TESTS = [
+        {
+            'url': 'https://echo360.org.uk/media/1d8392aa-a3e7-4e78-94cf-b6532c27208c/public',
+            'info_dict': {
+                'id': '3c7ae6e0-fa19-432d-aa21-c283b4276f2a',
+                'ext': 'mp4',
+                'title': '3-4 Force + moment + mechanics.mp4',
+                'duration': 4731.888,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+        {
+            'url': 'https://echo360.net.au/media/f04960a9-2efc-4b63-87b5-72e629081d15/public',
+            'info_dict': {
+                'id': '6098a147-2d65-40f3-b9e9-a0204afe450c',
+                'ext': 'mp4',
+                'title': 'EXSC634_Online_Workshop_Week_4.mp4',
+                'duration': 6659.72,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _call_api(self, host, video_id, media_id, session_token, **kwargs):
+        return self._download_json(
+            f'https://{host}/api/ui/echoplayer/public-links/{video_id}/media/{media_id}/player-properties',
+            video_id, headers={'Authorization': f'Bearer {session_token}'}, **kwargs)
+
+    def _replace_url_query(self, url, query_string):
+        if query_string is not None:
+            return urlunparse(urlparse(url)._replace(query=query_string))
+        return url
+
+    def _get_query_string(self, uri, query_strings):
+        uri_base = urlparse(uri)._replace(query='', fragment='').geturl()
```

Comment:
Import `update_url` from utils and you can instead use `update_url(url, query=query_string)` where the `_replace_url_query` method was needed.
(Remove `urllib.parse` from imports)
```suggestion
    def _get_query_string(self, uri, query_strings):
        uri_base = update_url(uri, query=None, fragment=None)
```